client/client.go

```diff
@@ -108,13 +126,28 @@ func (c *Client) SendRequest(method string, rawURL string, data url.Values,
 		}
 	}
 
-	if method == http.MethodPost {
-		valueReader = strings.NewReader(data.Encode())
+	//data is already processed and information will be added to u(the url) in the
+	//previous step. Now body will solely contain json payload 
+	if contentType == jsonContentType {
+		req, err = http.NewRequest(method, u.String(), bytes.NewBuffer(body))
+		if err != nil {
+			return nil, err
+		}		
+	} else {
+		//Here the HTTP POST methods which is not having json content type are processed
+		//All the values will be added in data and encoded (all body, query, path parameters)
+		if method == http.MethodPost {
```

Comment:
Going forward PUT method can be used for update operation which will have request body.